### PR TITLE
Add Makefile, configuration files and scripts

### PR DESCRIPTION
**Pull request by StefanJum**
Update the README with instructions on setting up, configuring, building and running the Python3 app with Unikraft. As part of it, add Makefile files, configuration files and scripts; they are referenced in the `README.md` file.

Current instructions cover the use of QEMU/KVM platform (both on x86_64 and on AArch64) and GCC. Firecraker/KVM, Xen, Linux platforms and Clang are not yet documented.

### CODE REVIEW

**Review by John-Ted — APPROVED** [review]
Reviewed-by: Ioan-Teodor Teugea ioan_teodor.teugea@stud.acs.upb.ro

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/.config
+/fs0/nginx/logs/
```
Comment (razvand) [inline .gitignore]: Copy-paste? For shame!

I would do a separate commit for ignoring output files.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+UK_ROOT ?= $(PWD)/.unikraft/unikraft
+UK_LIBS ?= $(PWD)/.unikraft/libs
+LIBS := $(UK_LIBS)/musl:$(UK_LIBS)/lwip:$(UK_LIBS)/python3:$(UK_LIBS)/libuuid
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+
```
Comment (razvand) [inline Makefile]:
```suggestion
```

File: Makefile.uk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$(eval $(call addlib,apppython3))
+
```
Comment (razvand) [inline Makefile.uk]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -1,49 +1,305 @@
...
-kraft build
-```
+* `build-essential` / `base-devel` / `@development-tools` (the meta-package that includes `make`, `gcc` and other development-related packages)
+* `sudo`
+* `flex`
+* `bison`
+* `git`
+* `wget`
+* `uuid-runtime`
+* `qemu-system-x86`
```
Comment (razvand) [inline README.md:17]:
```suggestion
* `qemu-system-x86`
* `qemu-system-arm`
```

File: README.md [hunk 5]
```diff
@@ -1,49 +1,305 @@
...
+  gcc-aarch64-linux-gnu \
+  libncurses-dev \
+  libyaml-dev \
+  flex \
+  bison \
+  git \
+  wget \
+  uuid-runtime \
+  qemu-kvm \
+  qemu-system-x86 \
```
Comment (razvand) [inline README.md:40]:
```suggestion
  qemu-system-x86 \
  qemu-system-arm \
```

File: README.md [hunk 6]
```diff
@@ -1,49 +1,305 @@
...
+     build  config-qemu-aarch64  config-qemu-x86_64  kraft.yaml  Makefile  Makefile.uk  minrootfs.tgz  README.md  run-qemu-aarch64  run-qemu-x86_64
+     ```
+
+  1. While inside the `python3/` directory, create the `fs0/` directory and extract the contents of `minrootfs.tgz`:
+
+     ```console
+     $ mkdir fs0/
+     $ cd fs0/
+     $ tar xvf ../minrootfs.tgz
+     $ cd ..
```
Comment (razvand) [inline README.md]:
```suggestion
     $ mkdir fs0/
     $ tar -C fs0/ -xvf minrootfs.tgz
```

File: .gitignore (hunk 1 above)
Comment (razvand) [reply .gitignore]: It should be `/fs0/`.

**Review by razvand — CHANGES_REQUESTED** [review]
@StefanJum, see my comments. I would also do a commit to rename `minrootfs.tgz` to `rootfs.tar.gz`. The `.tgz` extension is weird. And no reason to confuse people about the `min` part in `minrootfs`.

File: README.md [hunk 7]
```diff
@@ -1,49 +1,305 @@
...
+* `sgabios`
+* `gcc-aarch64-linux-gnu`
+
+GCC >= 8 is required to build Python3 on Unikraft.
 
-And finally, run the application:
+On Ubuntu/Debian or other `apt`-based distributions, run the following command to install the requirements:
+
+```console
+$ apt-get install -y --no-install-recommends \
```
Comment (razvand) [inline README.md]:
```suggestion
$ sudo apt install -y --no-install-recommends \
```

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,4 @@
+/.config
```
Comment (razvand) [inline .gitignore:1]:
```suggestion
/.config
/.config.old
```

File: README.md [hunk 9]
```diff
@@ -1,49 +1,305 @@
...
+     $ cd python3/
+
+     $ ls -F
+     build  config-qemu-aarch64  config-qemu-x86_64  kraft.yaml  Makefile  Makefile.uk  rootfs.tar.gz  README.md  run-qemu-aarch64  run-qemu-x86_64
+     ```
+
+  1. While inside the `python3/` directory, create the `fs0/` directory and extract the contents of `rootfs.tar.gz`:
+
+     ```console
+     $ mkdir fs0/
```
Comment (razvand) [inline README.md:78]:
```suggestion
     $ rm -fr fs0/
     $ mkdir fs0/
```

File: README.md [hunk 10]
```diff
@@ -1,49 +1,305 @@
...
+     $ tar -C fs0/ xvf ../rootfs.tar.gz
```
Comment (razvand) [inline README.md]:
```suggestion
     $ tar -C fs0/ -xvf rootfs.tar.gz
```

**Review by razvand — CHANGES_REQUESTED** [review]
@StefanJum, see my comments.

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
